allennlp/models/graph_parser.py

```diff
@@ -128,16 +128,20 @@ def __init__(self,
     @overrides
     def forward(self,  # type: ignore
                 tokens: Dict[str, torch.LongTensor],
+                metadata: List[Dict[str, Any]],
```

Comment:
I'd probably default to making this optional instead of moving the parameter, but I could go either way.

Reply:
How? By returning the `"tokens"` entry only if `metadata` is provided?

Reply:
Yes.